crates/test-programs/wasi-tests/src/bin/path_filestat.rs

```diff
@@ -117,8 +117,61 @@ unsafe fn test_path_filestat(dir_fd: wasi::Fd) {
     assert_eq!(stat.mtim, new_mtim, "mtim should not change");
     assert_eq!(stat.atim, old_atim, "atim should not change");
 
+    // Create a symlink
+    wasi::path_symlink("file", dir_fd, "symlink").expect("creating symlink to a file");
+
+    // Check path_filestat_set_times on the symlink itself
+    let mut sym_stat = wasi::path_filestat_get(dir_fd, 0, "file").expect("reading file stats");
+
+    let sym_new_mtim = sym_stat.mtim - 200;
+    wasi::path_filestat_set_times(
+        dir_fd,
+        0,
+        "symlink",
+        // on purpose: the syscall should not touch atim, because
+        // neither of the ATIM flags is set
+        sym_new_mtim,
+        sym_new_mtim,
+        wasi::FSTFLAGS_MTIM,
+    )
+    .expect("path_filestat_set_times should succeed on symlink");
+
+    sym_stat = wasi::path_filestat_get(dir_fd, 0, "symlink")
+        .expect("reading file stats after path_filestat_set_times");
+    assert_eq!(sym_stat.mtim, new_mtim, "mtim should change");
```

Comment:
I'm unclear on what this is testing. `new_mtim` is `stat.mtim - 100` from above. `sym_stat.mtim` is from the stat we just got after setting the time to `sym_new_mtim` which is `sym_stat.mtim - 200`. It's not clear why those should be equal. Also, the message here "mtim should change" doesn't seem to fit using `assert_eq` to check that two things are the same.

Reply:
Oh you're right! It was meant to be `sys_new_mtim` rather than `new_mtim`. Now it makes we wonder why it passed the tests at all.

Reply:
OK, I've fixed it now in the latest commit.